pkg/auth/authmap_writer.go

```diff
@@ -77,3 +80,28 @@ func (r *authMapWriter) Delete(key authKey) error {
 
 	return nil
 }
+
+func (r *authMapWriter) DeleteIf(predicate func(key authKey, info authInfo) bool) error {
+	var entriesToDelete []authKey
+	all, err := r.All()
+	if err != nil {
+		return fmt.Errorf("failed to get all entries: %w", err)
+	}
+	for k, v := range all {
+		if predicate(k, v) {
+			entriesToDelete = append(entriesToDelete, k)
+		}
+	}
+
+	for _, key := range entriesToDelete {
```

Comment:
Same question as earlier. What's the benefit of the more complicated double looping and extra data structure?

Reply:
@joamaki thanks - fixed!